spec/octopus/serverspec/type/octopus_deploy_project_group_spec.rb

```diff
@@ -0,0 +1,112 @@
+require 'spec_helper'
+require 'webmock/rspec'
+
+describe OctopusDeployProjectGroup do
+
+  let(:runner) { double ("runner")}
+
+  it "throws if `serverUrl` not supplied" do
+    expect { OctopusDeployProjectGroup.new(nil, "someapikey", "my new projectgroup") }.
+        to raise_error(/serverUrl/)
+  end
+
+  it "throws if `apiKey` not supplied" do
+    expect { OctopusDeployProjectGroup.new("https://someserver.com", nil, "my new projectgroup") }.
+        to raise_error(/apiKey/)
+  end
+
+  it "throws if `ProjectGroupName` not supplied" do
+    expect { OctopusDeployProjectGroup.new("https://someserver.com", "API-kllkjhasdkljhasdfkjsafd", nil) }.
+        to raise_error(/projectgroup_name/)
+  end
+
+  ex_does_not_support_spaces = File.read('spec/octopus/serverspec/json/apidoesnotsupportspaces.json')
```

Comment:
not enthused on the name...

Reply:
The filename? I'm not overly attached to it, can change that if necessary.

Reply:
the `ex_does_not_support_spaces`, and the other 3 variables.